### PR TITLE
rgw_sal_motr : [CORTX-34168] Modified response fields of DeleteObjects API

### PR DESCRIPTION
**Pull request by priyanka25081999**
Problem Statement: 
DELETE Objects API should return Deleted key, DeleteMarker and DeleteMarkerVersionId in response when a delete marker is created for an object in a versioned bucket

Solution:
Set the `delete_marker` variable to `true` to get "Deleted key, DeleteMarker and DeleteMarkerVersionId" in the response. 

Signed-off-by: Priyanka Salunke <priyanka.s.salunke@seagate.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -2137,6 +2137,7 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
       source->gen_rand_obj_instance_name();
       std::string del_marker_ver_id = source->get_instance();
       result.version_id = del_marker_ver_id;
+      source->delete_marker = true;
```
Comment (sachitanands) [inline src/rgw/rgw_sal_motr.cc:2140]: at line 2136 also we are setting delete_marker on result object. 
Please check if we need to set this boolean value for both result & source object.

Comment (priyanka25081999) [reply src/rgw/rgw_sal_motr.cc:2140]: Yes
`Results` structure contains delete-marker field to identify that delete-marker entry is present for an object and should be visible when we perform list-object-versions.
Here, we are setting the delete-marker to true to visible the response fields such as "DeleteMarker, DeleteMarkerVersionId" in the DeleteObjects API response.

**Review by cdeshmukh — APPROVED** [review]

**Review by sachitanands — APPROVED** [review]

**Review by shraddhaghatol — APPROVED** [review]

**Comment by priyanka25081999** [comment]
1N k8s Auto-triggered deployment is successful - https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Generic/job/cortx-rgw/215/

1N k8s manual deployment is successful - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/12532/

**Comment by priyanka25081999** [comment]
Sanity job is successful - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/S3Server/job/S3-Compatibility-Test-Ceph-RGW/243/ [PASSED]
image

**Pull request merged**